src/services/azureFunctionsService.ts

```diff
@@ -68,16 +68,34 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 			return;
 		}
 		let projectFile = await azureFunctionUtils.getAzureFunctionProject();
+		let newHostProjectFile: azureFunctionUtils.IFileFunctionObject;
+		let hostFile: string;
+
 		if (!projectFile) {
 			let projectCreate = await vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened,
 				LocalizedConstants.createProject, LocalizedConstants.learnMore);
 			if (projectCreate === LocalizedConstants.learnMore) {
 				vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(constants.sqlBindingsDoc));
+				return;
 			} else if (projectCreate === LocalizedConstants.createProject) {
-				// start the create azure function project flow
-				await azureFunctionApi.createFunction({});
+				try {
+					// start the create azure function project flow
+					// create a watcher to see if the host file is created after the flow is complete
+					newHostProjectFile = await azureFunctionUtils.waitForNewHostFile();
+					const timeoutForHostFile = timeoutPromise(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+					await azureFunctionApi.createFunction({});
+					hostFile = await Promise.race([newHostProjectFile.filePromise, timeoutForHostFile]);
+					if (hostFile) {
+						// start the add sql binding flow
+						projectFile = await azureFunctionUtils.getAzureFunctionProject();
+					}
+				} catch (error) {
+					vscode.window.showErrorMessage(error.message);
```

Comment:
Generally we should include header text saying where the error is from `Error creating new Azure Function : ...`

Also it's good practice to not assume that a caught error is an error object. Most of the time it should be but sometimes other libraries throw strings instead. So you should do `error.message ?? error`